gen/nvml/lib_test.go

```diff
@@ -134,6 +134,43 @@ func TestLookupFromDefault(t *testing.T) {
 	}
 }
 
+func TestLoadAndCloseNesting(t *testing.T) {
+	var numOpens, numCloses int
+
+	dl := &dynamicLibraryMock{
+		OpenFunc: func() error {
+			numOpens++
+			return nil
+		},
+		CloseFunc: func() error {
+			numCloses++
+			return nil
+		},
+	}
+
+	defer setNewDynamicLibraryDuringTest(dl)()
+	defer resetLibrary()
+	GetLibrary()
+
+	// When calling load twice, the library was only opened once
```

Comment:
Could we add the following initial check too:
```
	// When calling close before opening the library nothing happens.
	require.Equal(t, 0, len(dl.calls.Close))
	require.Nil(t, libnvml.close())
	require.Equal(t, 0, len(dl.calls.Close))
```
to ensure that we don't try to close a nil library?

Reply:
Integrating ...